src/kubernetes/reflector.rs

```diff
@@ -112,7 +112,7 @@ where
                 let val = select! {
                     // If we got a maintenance request - perform the
                     // maintenance.
-                    _ = maintenance_request.unwrap(), if maintenance_request.is_some() => {
+                    _ = async { maintenance_request.unwrap().await }, if maintenance_request.is_some() => {
```

Comment:
I'm curious what you think might be the difference here.

Reply:
My only idea is that it has to do with whatever the `select!` macro is adding combined with a possible borrow vs ownership change.